business/namespaces.go

```diff
@@ -556,15 +557,23 @@ func (in *NamespaceService) GetNamespaceByCluster(ctx context.Context, namespace
 			for cl := range in.userClients {
 				project, err2 = in.userClients[cl].GetProject(namespace)
 				if err2 == nil {
-					result = models.CastProject(*project)
+					result = models.CastProject(*project, cluster)
 					break
 				}
 			}
 			if err2 != nil {
 				return nil, err2
 			}
+		} else {
```

Comment:
here is one of the key changes, the issue described here https://github.com/kiali/kiali/pull/6106  is resolved now